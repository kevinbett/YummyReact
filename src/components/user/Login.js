import React, { Component } from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import RaisedButton from 'material-ui/RaisedButton';
import FlatButton from 'material-ui/FlatButton';
import Dialog from 'material-ui/Dialog';
import TextField from 'material-ui/TextField';
import {Tabs, Tab} from 'material-ui/Tabs';
import SwipeableViews from 'react-swipeable-views';
import {notify} from 'react-notify-toast';
import axiosInstance from '../../Axios';
import Register from './Register';
import { Link } from 'react-router-dom';
import red500 from 'material-ui/styles/colors';
import LinearProgress from 'material-ui/LinearProgress';

const styles = {
    headline: {
      fontSize: 24,
      paddingTop: 16,
      marginBottom: 12,
      fontWeight: 600,
    },
    slide: {
      padding: 10,
    },
    tabItemContainer: {
		background: 'none'
	},
  };


class Login extends Component {
    constructor(props) {
        super(props);
        this.state = {
            username: '',
            password: '',
            error: '',
            slideIndex: 0,
            open: false,
        }
        this.handleCancel = this.handleCancel.bind(this);
        this.handleResetPassword = this.handleResetPassword.bind(this);        
    }
    handleCancel(event) {
        event.preventDefault();
        this.props.history.push('/');
    };
    handleOpen = () => {
        this.setState({open: true});
    };
    handleChange = (value) => {
        this.setState({
          slideIndex: value,
        });
      };
    handleResetChange = event => {
        this.setState({ username: event.target.value , email: event.target.value});       
    }
    handleClose = () => {
        this.setState({open: false, error: '', message:''});
    };
    handleResetPassword(event){ 
        event.preventDefault();  
        var payload = {
            "username": this.state.username,
            "email": this.state.email
        }
        axiosInstance.post('/auth/reset-password/', payload)
            .then(response => {
                this.setState({message: response.data.message});
            })
            .catch(error => {
                this.setState({error: error.response.data.message})
            })
      }

    render() {
        const actions = [
            <FlatButton
              label="Submit"
              primary={true}
              keyboardFocused={true}
              onClick={(event) => {
                  this.handleResetPassword(event);
                  <LinearProgress mode="indeterminate" />
                  }}
            />,
            <FlatButton
              label="Close"
              primary={false}
              keyboardFocused={false}
              onClick={this.handleClose}
            />,
          ];
        return (

            <div className="Login">
                <MuiThemeProvider>
                <Dialog
                    title="Reset Password"
                    actions={actions}
                    modal={false}
                    open={this.state.open}
                    onRequestClose={this.handleClose}
                    >
                    { this.state.error?
                    <label color={red500}> {this.state.error}</label>: ''}
                    { this.state.message?
                    <label color={red500}> {this.state.message}</label>: ''}
                    <form onSubmit={this.handleSubmit}>
                        <TextField
                        hintText="Enter Username"
                        floatingLabelText="Username"
                        fullWidth={true}
                        onChange={(event, newValue) =>
                                            this.setState({ username: newValue })} />
                        <br/>
                        <TextField
                        hintText="Enter Email"
                        floatingLabelText="Enter Email"
                        fullWidth={true}                        
                        onChange={(event, newValue) =>
                                            this.setState({ email: newValue })} />
                    </form>
                </Dialog> 
                    <div>
                        <Tabs
                        onChange={this.handleChange}
                        tabItemContainerStyle={styles.tabItemContainer}
                        value={this.state.slideIndex}
                        >
                        <Tab label="Login" style={{ color: 'black' }} value={0} />
                        <Tab label="Register" style={{ color: 'black' }} value={1} />
                        </Tabs>
                        <SwipeableViews
                        index={this.state.slideIndex}
                        onChangeIndex={this.handleChange}
                        >
                        <div>
                            <form>
                            <TextField
                                hintText="Enter your Username"
                                floatingLabelText="Username"
                                onChange={(event, newValue) =>
                                    this.setState({ username: newValue })} />
                            <br />
                            <TextField
                                type="Enter password"
                                hintText="Enter your Password"
                                floatingLabelText="Password"
                                type='password'
                                onChange={(event, newValue) =>
                                    this.setState({ password: newValue })} />
                            <br />
                            <RaisedButton type='submit' label="Submit" secondary={true}
                                style={style} onClick={(event) => this.handleClick(event)} />
                            <br/><br/> 
                            <p>Forgot password? <FlatButton onClick={(event) => this.handleOpen()}>Reset</FlatButton></p>
                        </form>
                        </div>
                        <div style={styles.slide}>
                            <Register/>
                        </div>
                        </SwipeableViews>
                    </div>
                </MuiThemeProvider>
            </div>
        );
    }
    handleClick(event) {
        event.preventDefault();
        var payload = {
            "username": this.state.username,
            "password": this.state.password
        }
        axiosInstance.post('auth/login/', payload)
            .then(response => {
                window.localStorage.setItem("token", response.data.token)
                window.localStorage.setItem("username", response.data.username)
                window.localStorage.setItem("logged_in", true)
                window.location.assign('/dashboard/categories')
                }
            )
            .catch(error => {
                window.location.assign("/login");                
                notify.show(error.response.data.message, 'error', 4000)
            })
    }

}

const style = {
    marginBottom: 0,
};
export default Login;