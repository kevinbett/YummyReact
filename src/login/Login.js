import React, { Component } from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import RaisedButton from 'material-ui/RaisedButton';
import TextField from 'material-ui/TextField';
import {Tabs, Tab} from 'material-ui/Tabs';
import SwipeableViews from 'react-swipeable-views';
import {notify} from 'react-notify-toast';
import axios from 'axios';
import Register from './register';

const styles = {
    headline: {
      fontSize: 24,
      paddingTop: 16,
      marginBottom: 12,
      fontWeight: 600,
    },
    slide: {
      padding: 10,
    },
    tabItemContainer: {
		background: 'none'
	},
  };


class Login extends Component {
    constructor(props) {
        super(props);
        this.state = {
            username: '',
            password: '',
            error: '',
            slideIndex: 0
        }
        this.handleCancel = this.handleCancel.bind(this);
    }
    handleCancel(event) {
        event.preventDefault();
        this.props.history.push('/');
    }
    handleChange = (value) => {
        this.setState({
          slideIndex: value,
        });
      };
    render() {
        return (

            <div className="Login">
                <MuiThemeProvider>
                    <div>
                        <Tabs
                        onChange={this.handleChange}
                        tabItemContainerStyle={styles.tabItemContainer}
                        value={this.state.slideIndex}
                        >
                        <Tab label="Login" style={{ color: 'black' }} value={0} />
                        <Tab label="Register" style={{ color: 'black' }} value={1} />
                        </Tabs>
                        <SwipeableViews
                        index={this.state.slideIndex}
                        onChangeIndex={this.handleChange}
                        >
                        <div>
                            <form>
                            <TextField
                                hintText="Enter your Username"
                                floatingLabelText="Username"
                                onChange={(event, newValue) =>
                                    this.setState({ username: newValue })} />
                            <br />
                            <TextField
                                type="password"
                                hintText="Enter your Password"
                                floatingLabelText="Password"
                                onChange={(event, newValue) =>
                                    this.setState({ password: newValue })} />
                            <br />
                            <RaisedButton type='submit' label="Submit" secondary={true}
                                style={style} onClick={(event) => this.handleClick(event)} />
                        </form>
                        </div>
                        <div style={styles.slide}>
                            <Register/>
                        </div>
                        </SwipeableViews>
                    </div>
                </MuiThemeProvider>
            </div>
        );
    }
    handleClick(event) {
        event.preventDefault();
        var apiBaseUrl = "http://localhost:5000/api/auth/";
        var payload = {
            "username": this.state.username,
            "password": this.state.password
        }
        axios.post(apiBaseUrl + 'login/', payload)
            .then(response => {
                window.localStorage.setItem("token", response.data.token)
                window.localStorage.setItem("username", response.data.username)
                if (response.status === 200) {
                    window.location.assign('/dashboard/categories')
                }
                else {
                    window.location.assign("/login");
                }
            })
            .catch(error => {
                notify.show(error.response.data.message, 'error', 4000)
            })
    }

}

const style = {
    marginBottom: 0,
};
export default Login;