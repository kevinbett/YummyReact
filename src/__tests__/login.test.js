import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import Login from '../login/Login';

describe('Login Component', () => {
    const wrapper = shallow(<Login />);
    const preventDefault = jest.fn();
    it('renders without crashing', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
    it('renders state initially', () => {
        expect(wrapper.state().username).toEqual('');
        expect(wrapper.state().password).toEqual('');
    })
});