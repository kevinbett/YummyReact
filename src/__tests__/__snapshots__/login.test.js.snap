// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component renders without crashing 1`] = `
<div
  className="Login"
>
  <MuiThemeProvider>
    <Dialog
      actions={
        Array [
          <FlatButton
            disabled={false}
            fullWidth={false}
            keyboardFocused={true}
            label="Submit"
            labelPosition="after"
            labelStyle={Object {}}
            onClick={[Function]}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            primary={true}
            secondary={false}
          />,
          <FlatButton
            disabled={false}
            fullWidth={false}
            keyboardFocused={false}
            label="Close"
            labelPosition="after"
            labelStyle={Object {}}
            onClick={[Function]}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            primary={false}
            secondary={false}
          />,
        ]
      }
      autoDetectWindowHeight={true}
      autoScrollBodyContent={false}
      modal={false}
      onRequestClose={[Function]}
      open={false}
      repositionOnUpdate={true}
      title="Reset Password"
    >
      <form>
        <TextField
          disabled={false}
          floatingLabelFixed={false}
          floatingLabelText="Username"
          fullWidth={true}
          hintText="Enter Username"
          multiLine={false}
          onChange={[Function]}
          rows={1}
          type="text"
          underlineShow={true}
        />
        <br />
        <TextField
          disabled={false}
          floatingLabelFixed={false}
          floatingLabelText="Enter Email"
          fullWidth={true}
          hintText="Enter Email"
          multiLine={false}
          onChange={[Function]}
          rows={1}
          type="text"
          underlineShow={true}
        />
      </form>
    </Dialog>
    <div>
      <Tabs
        initialSelectedIndex={0}
        onChange={[Function]}
        tabItemContainerStyle={
          Object {
            "background": "none",
          }
        }
        value={0}
      >
        <Tab
          label="Login"
          style={
            Object {
              "color": "black",
            }
          }
          value={0}
        />
        <Tab
          label="Register"
          style={
            Object {
              "color": "black",
            }
          }
          value={1}
        />
      </Tabs>
      <ReactSwipableView
        animateHeight={false}
        animateTransitions={true}
        axis="x"
        disableLazyLoading={false}
        disabled={false}
        enableMouseEvents={false}
        hysteresis={0.6}
        ignoreNativeScroll={false}
        index={0}
        onChangeIndex={[Function]}
        resistance={false}
        springConfig={
          Object {
            "delay": "0s",
            "duration": "0.35s",
            "easeFunction": "cubic-bezier(0.15, 0.3, 0.25, 1)",
          }
        }
        threshold={5}
      >
        <div>
          <form>
            <TextField
              disabled={false}
              floatingLabelFixed={false}
              floatingLabelText="Username"
              fullWidth={false}
              hintText="Enter your Username"
              multiLine={false}
              onChange={[Function]}
              rows={1}
              type="text"
              underlineShow={true}
            />
            <br />
            <TextField
              disabled={false}
              floatingLabelFixed={false}
              floatingLabelText="Password"
              fullWidth={false}
              hintText="Enter your Password"
              multiLine={false}
              onChange={[Function]}
              rows={1}
              type="password"
              underlineShow={true}
            />
            <br />
            <RaisedButton
              disabled={false}
              fullWidth={false}
              label="Submit"
              labelPosition="after"
              onClick={[Function]}
              primary={false}
              secondary={true}
              style={
                Object {
                  "marginBottom": 0,
                }
              }
              type="submit"
            />
            <br />
            <br />
            <p>
              Forgot password? 
              <FlatButton
                disabled={false}
                fullWidth={false}
                labelPosition="after"
                labelStyle={Object {}}
                onClick={[Function]}
                onKeyboardFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onTouchStart={[Function]}
                primary={false}
                secondary={false}
              >
                Reset
              </FlatButton>
            </p>
          </form>
        </div>
        <div
          style={
            Object {
              "padding": 10,
            }
          }
        >
          <Register />
        </div>
      </ReactSwipableView>
    </div>
  </MuiThemeProvider>
</div>
`;
